pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -139,7 +139,24 @@ func NewController(opt Options) *Controller {
 }
 
 func (c *Controller) addRegistry(registry serviceregistry.Instance, stop <-chan struct{}) {
-	c.registries = append(c.registries, &registryEntry{Instance: registry, stop: stop})
+	added := false
+	if registry.Provider() == provider.Kubernetes {
+		for i, r := range c.registries {
+			if r.Provider() != provider.Kubernetes {
+				// insert the registry in the position of the first non kubernetes registry
+				newRegistries := make([]*registryEntry, len(c.registries)+1)
```

Comment:
nit: can we use slices.Insert to avoid this complex code

Reply:
good to know that function.